build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java

```diff
@@ -189,9 +189,24 @@ private void createTempNpmrc(Path workingDir, List<String> commandArgs) throws I
             npmrcBuilder.append("proxy = ").append(this.npmProxy).append("\n");
         }
 
+        //Update Auth property for newer npm versions
+        if( this.npmDriver.compareVersionTo(workingDir.toFile(), "8.19") >= 0){
+            try (ArtifactoryManager artifactoryManager = artifactoryManagerBuilder.build()) {
+                String authProp = npmAuth.getProperty("_auth");
+
+                String newAuthKey = artifactoryManager.getUrl();
+                if (!StringUtils.endsWith(newAuthKey, "/")) {
+                    newAuthKey += "/";
+                }
+                newAuthKey += "api/npm/:_auth";
+                newAuthKey = newAuthKey.replaceAll("^http(s)?:","") + ":_auth";
+                npmAuth.setProperty(newAuthKey, authProp);
+                npmAuth.remove("_auth");
+            }
+        }
```

Comment:
* To make the code easier to read. let move the above code into a new privated method named `handleNpmCompatibility`
* Our coding conversion include having a single white-space after //
* To align with our coding conversions, the following if statement - 
`if( this.npmDriver.compareVersionTo(workingDir.toFile(), "8.19") >= 0){`
should be modified to - 
`if (this.npmDriver.compareVersionTo(workingDir.toFile(), "8.19") >= 0) {`
(notice the white-spacing conventions)
* Let's add a unit test for this functionality in `NpmBuildInfoExtractorTest`.

Reply:
The added test case is 99% the same as the NpmExtractorTest.npmCiTest, which under the NpmInstallCi and subsequent .execute does in-fact call the NPMBuildInfoExtractor method. There is nothing specific for the test to check except that it should pass for both the Node 15 and Node 20 now found in the integrationTests.yml. I'm unable to execute this fully on my side as I do not have Admin access to an Artifactory system and the open source does not have NPM enabled.